source/Halibut/Transport/SecureListener.cs

```diff
@@ -248,18 +254,25 @@ await using (Try.CatchingErrorOnDisposal(ssl, ex => log.WriteException(EventType
 
                     if (Authorize(thumbprint, clientName))
                     {
-                        connectionAuthorizedAndObserved = true;
-                        connectionsObserver.ConnectionAccepted(true);
-                        tcpClientManager.AddActiveClient(thumbprint, client);
-                        errorEventType = EventType.Error;
-                        await ExchangeMessages(ssl).ConfigureAwait(false);
+                        using (var _ = authorizedTcpConnectionsLimiter.ClaimAuthorizedTcpConnection(thumbprint))
```

Comment:
Let's not forget about a feature flag so we can disable this if something goes wrong.

Reply:
Does Halibut have feature flags?

Reply:
I hope so :).  @LukeButters , how did you ship significant changes to Halibut?

Reply:
We update Halibut with a new bool/int etc which has the same effective value as what halibut always did (thus toggled off in Halibut)

In Tentacle and Octopus we toggle that to on/off depending on feature flags.

Eventually when we are super sure it works we default it to on in Halibut and eventually remove the feature flags.

Reply:
Changed to a nullable int, so `null` performs no limiting (current behaviour)